src/veritysetup/veritysetup.c

```diff
@@ -41,6 +53,11 @@ static int help(void) {
         return 0;
 }
 
+static int block_size_is_valid(uint64_t block_size) {
+        return (block_size % 512) == 0 && (block_size & (block_size-1)) != 0 &&
+               (block_size >= 512) && (block_size <= (512 * 1024));
```

Comment:
you can drop at least () here.

maybe add a comment about what `(block_size & (block-size-1)) != 0` actually is supposed to check, it's not obvious...